plugins/source/scheduler_dfs.go

```diff
@@ -169,6 +169,12 @@ func (p *Plugin) resolveResourcesDfs(ctx context.Context, table *schema.Table, c
 				if resolvedResource == nil {
 					return
 				}
+				tableMetrics := p.metrics.TableClient[table.Name][client.ID()]
```

Comment:
Maybe this lookup should happen inside the `if`, because that's the only place it's used?

Reply:
Good catch.